packages/cli/src/verify.ts

```diff
@@ -0,0 +1,80 @@
+import { forge, getRpcUrl } from "@latticexyz/common/foundry";
+import { Hex, createWalletClient, getCreate2Address, http } from "viem";
+import { salt } from "./deploy/common";
+import { ensureDeployer } from "./deploy/ensureDeployer";
+import { privateKeyToAccount } from "viem/accounts";
+import { MUDError } from "@latticexyz/common/errors";
+import { getWorldFactoryContracts } from "./deploy/ensureWorldFactory";
+import { Contract } from "./deploy/ensureContract";
+
+type VerifyOptions = {
+  foundryProfile?: string;
+  systems: Contract[];
+  modules: Contract[];
+  worldAddress?: Hex;
+};
+
+async function verifyContract(
+  foundryProfile: string | undefined,
+  deployerAddress: Hex,
+  contract: Contract,
+  rpc: string,
+  cwd?: string,
+) {
+  if (!contract.label) {
```

Comment:
can we do this at the type level instead of runtime?

Reply:
We could use an `as` type assertion, or rework the whole `Contract` type this is using

Reply:
could probably do something like `contract: Contract & Required<Pick<Contract, "label">>